node-js-basics-101/node_modules/asynquence-contrib/contrib-es6.src.js

```diff
@@ -0,0 +1,2360 @@
+
+		return template;
+	};
+
+	ASQ.react.extend = function $$extend(name,build) {
+		extensions[name] = build;
+
+		return ASQ.react;
+	};
+
+})();
+// "react" helpers
+(function IIFE(){
+
+	var Ar = ASQ.react;
+
+	Ar.of = function $$react$of() {
+		function reactor(next) {
+			if (!started) {
+				started = true;
+				if (args.length > 0) {
+					args.shift().val(function val(){
+						next.apply(ø,arguments);
+						if (args.length > 0) {
+							args.shift().val(val);
+						}
+					});
+				}
+			}
+		}
+
+		var started, args = ARRAY_SLICE.call(arguments)
+			.map(function wrapper(arg){
+				if (!ASQ.isSequence(arg)) arg = ASQ(arg);
+				return arg;
+			});
+
+		return Ar(reactor);
+	};
+
+	Ar.all = Ar.zip = makeReactOperator(/*buffer=*/true);
+	Ar.allLatest = makeReactOperator(/*buffer=false*/);
+	Ar.latest = Ar.combineLatest = makeReactOperator(/*buffer=*/false,/*keep=*/true);
+
+	Ar.any = Ar.merge = function $$react$any(){
+		function reactor(next,registerTeardown){
+			function processSequence(def){
+				function trigger(){
+					var args = ASQ.messages.apply(ø,arguments);
+					// still observing sequence-streams?
+					if (seqs && seqs.length > 0) {
+						// fire off reactive sequence instance
+						next.apply(ø,args);
+					}
+					// keep sequence going
+					return args;
+				}
+
+				// sequence-stream event listener
+				def.seq.val(trigger);
+			}
+
+			// observe all sequence-streams
+			seqs.forEach(processSequence);
+
+			// listen for stop() of reactive sequence
+			registerTeardown(function $$teardown(){
+				seqs = null;
+			});
+		}
+
+		// observe all sequence-streams
+		var seqs = tapSequences.apply(null,arguments);
+
+		if (seqs.length == 0) return;
+
+		return Ar(reactor);
+	};
+
+	Ar.distinct = function $$react$distinct(seq){
+		return Ar.filter(seq,makeDistinctFilterer(/*keepAll=*/true));
+	};
+
+	Ar.distinctConsecutive = Ar.distinctUntilChanged = function $$react$distinct$consecutive(seq) {
+		return Ar.filter(seq,makeDistinctFilterer(/*keepAll=*/false));
+	};
+
+	Ar.filter = function $$react$filter(seq,filterer){
+		function reactor(next,registerTeardown) {
+			function trigger(){
+				var messages = ASQ.messages.apply(ø,arguments);
+
+				if (filterer && filterer.apply(ø,messages)) {
+					// fire off reactive sequence instance
+					next.apply(ø,messages);
+				}
+
+				// keep sequence going
+				return messages;
+			}
+
+			// sequence-stream event listener
+			def.seq.val(trigger);
+
+			// listen for stop() of reactive sequence
+			registerTeardown(function $$teardown(){
+				def = filterer = null;
+			});
+		}
+
+		// observe sequence-stream
+		var def = tapSequences(seq)[0];
+
+		if (!def) return;
+
+		return Ar(reactor);
+	};
+
+	Ar.fromObservable = function $$react$from$observable(obsv){
+		function reactor(next,registerTeardown){
+			// process buffer (if any)
+			buffer.forEach(next);
+			buffer.length = 0;
+
+			// start non-buffered notifications?
+			if (!buffer.complete) {
+				notify = next;
```

Comment:
__[jshint]__
* Reassignment of 'notify', which is is a function. Use 'var' or 'let' to declare bindings that may change.

<!-- Comment by [SideCI](https://sideci.com) -->